testApp/src/App.svelte

```diff
@@ -0,0 +1,122 @@
+<!-- App.svelte -->
+<script lang="ts">
+	import { onMount } from 'svelte';
+	import Gallery from './Components/Gallery.svelte';
+	import MovieDetail from './Components/MovieDetail.svelte';
+	import NavBar from './Components/NavBar.svelte';
+	import '/global.css';
+	import type { Movie } from '/Interfaces/Movie';
+	import type { Franchise } from '/Interfaces/Franchise';
+
+	export let bearer: string;
+
+	let movies: Movie[] = [];
+	let currentIndex: number = -1;
+	let showDetail: boolean = false;
+	let currentMovie: Movie = null;
+	let franchiseId: number = 17; // Default to Star Wars
+
+	async function fetchFranchiseData(id: number) {
+		try {
+			const response = await fetch(`https://api4.thetvdb.com/v4/lists/${id}/extended`, {
+				method: 'GET',
+				headers: {
+					'accept': 'application/json',
+					'Authorization': bearer
+				}
+			});
+
+			if (!response.ok) {
+				console.error('Failed to fetch franchise data:', response.statusText);
+				return;
+			}
+
+			const data = await response.json();
+			const franchise: Franchise = data.data;
+
+			const moviePromises = franchise.entities
+					.filter(entity => entity.movieId)
+					.map(async entity => {
+						const movieResponse = await fetch(`https://api4.thetvdb.com/v4/movies/${entity.movieId}`, {
+							method: 'GET',
+							headers: {
+								'accept': 'application/json',
+								'Authorization': bearer
+							}
+						});
+
+						if (!movieResponse.ok) {
+							console.error(`Failed to fetch movie data for ID ${entity.movieId}:`, movieResponse.statusText);
+							return null;
+						}
+
+						const movieData = await movieResponse.json();
+						return movieData.data;
+					});
+
+			const movieResults = await Promise.all(moviePromises);
+			movies = movieResults.filter(movie => movie !== null);
+		} catch (error) {
+			console.error('Error fetching data:', error);
```

Comment:
I've changed this so that when no movies are found it will display a message that no movies were found and the app is still usable besides no movies loaded:

https://github.com/BowenGeense/MovieShowcase/pull/3/files#diff-1e1270da740e81dd13f8a65057582b0fce8a5fb3817ef913f3e90d7f82c4e9df

Is this a proper solution to the problem ?

Reply:
It is!